File: app/src/main/java/com/nandaadisaputra/projectakhir/MainActivity.java
package com.nandaadisaputra.projectakhir;

import android.content.Intent;
import android.os.Bundle;
import android.provider.Settings;
import android.view.Menu;
import android.view.MenuItem;

import androidx.appcompat.app.ActionBarDrawerToggle;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.drawerlayout.widget.DrawerLayout;
import androidx.fragment.app.Fragment;
import com.google.android.material.navigation.NavigationView;
import com.nandaadisaputra.projectakhir.fragment.AboutFragment;

public class MainActivity extends AppCompatActivity {
    private Fragment pageContent = new AboutFragment();
    private String title = "Aplikasi Mobile GIS";
    private String KEY_FRAGMENT;
    private String KEY_TITLE;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final Toolbar toolbar = findViewById(R.id.main_toolbar);
        final DrawerLayout drawerLayout = findViewById(R.id.main_drawer);

        ActionBarDrawerToggle drawerToggle = new ActionBarDrawerToggle(this, drawerLayout, toolbar, R.string.app_name, R.string.app_name);
        drawerLayout.addDrawerListener(drawerToggle);
        drawerToggle.syncState();

        NavigationView navigationView = findViewById(R.id.main_navigation);
        navigationView.setNavigationItemSelectedListener(menuItem -> {

            switch (menuItem.getItemId()) {
                case R.id.menu_tentang:
                    pageContent = new AboutFragment();
                    title = "Tentang SiPAI";
                    break;
                case R.id.menu_keluar:
                    finish();
                    System.exit(1);
                    break;
            }

            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, pageContent).commit();
            toolbar.setTitle(title);
            drawerLayout.closeDrawers();
            return true;
        });
        if (savedInstanceState == null) {
            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, pageContent).commit();
            toolbar.setTitle(title);
        } else {
            pageContent = getSupportFragmentManager().getFragment(savedInstanceState, KEY_FRAGMENT);
            title = savedInstanceState.getString(KEY_TITLE);

            getSupportFragmentManager().beginTransaction().replace(R.id.fragment_container, pageContent).commit();
            toolbar.setTitle(title);
        }
    }

    @Override
    protected void onSaveInstanceState(Bundle outState) {
        outState.putString(KEY_TITLE, title);
        getSupportFragmentManager().putFragment(outState, KEY_FRAGMENT, pageContent);
        super.onSaveInstanceState(outState);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(R.menu.main_menu, menu);
        return super.onCreateOptionsMenu(menu);
    }


    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.action_change_settings) {
            Intent mIntent = new Intent(Settings.ACTION_LOCALE_SETTINGS);
            startActivity(mIntent);
        }
        return super.onOptionsItemSelected(item);
    }

}

File: app/src/main/java/com/nandaadisaputra/projectakhir/activity/MainActivity.kt
package com.nandaadisaputra.projectakhir.activity

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.ActionBarDrawerToggle
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.drawerlayout.widget.DrawerLayout
import androidx.fragment.app.Fragment
import com.google.android.material.navigation.NavigationView
import com.nandaadisaputra.projectakhir.R
import com.nandaadisaputra.projectakhir.fragment.AboutFragment
import com.nandaadisaputra.projectakhir.network.SharedPrefManager
import org.jetbrains.anko.*
import kotlin.system.exitProcess

class MainActivity : AppCompatActivity() {
    private var pageContent: Fragment? = AboutFragment()
    private var title: String? = "Aplikasi Mobile GIS"
    private val KEY_FRAGMENT: String? = null
    private val KEY_TITLE: String? = null
    var sharedPrefManager: SharedPrefManager? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        sharedPrefManager = SharedPrefManager(this)
        val toolbar = findViewById<Toolbar>(R.id.main_toolbar)
        val drawerLayout = findViewById<DrawerLayout>(R.id.main_drawer)
        val drawerToggle = ActionBarDrawerToggle(this, drawerLayout, toolbar, R.string.app_name, R.string.app_name)
        drawerLayout.addDrawerListener(drawerToggle)
        drawerToggle.syncState()
        val navigationView = findViewById<NavigationView>(R.id.main_navigation)
        navigationView.setNavigationItemSelectedListener { menuItem: MenuItem ->
            when (menuItem.itemId) {
                R.id.menu_tentang -> {
                    pageContent = AboutFragment()
                    title = "Marketplace"
                }
                R.id.menu_keluar -> {
                    alert ("Apakah anda ingin logout ?"){
                        noButton {
                            toast("Anda tidak jadi Keluar")
                            startActivity(intentFor<MainActivity>())
                            finish()
                        }
                        yesButton {
                            if(sharedPrefManager?.sPSudahLogin!!) {
                                sharedPrefManager?.saveSPBoolean(SharedPrefManager.SP_SUDAH_LOGIN, false)
                                startActivity(Intent(this@MainActivity, LoginActivity::class.java)
                                        .addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP or Intent.FLAG_ACTIVITY_NEW_TASK))
                                super.onBackPressed()
                            }else {
                                toast("Maaf Aplikasi Tugas Akhir Mobile Program belum sempurna")
                            }
                        }
                    }.show()
                }
            }
            supportFragmentManager.beginTransaction().replace(R.id.fragment_container, pageContent!!).commit()
            toolbar.title = title
            drawerLayout.closeDrawers()
            true
        }
        if (savedInstanceState == null) {
            supportFragmentManager.beginTransaction().replace(R.id.fragment_container, pageContent!!).commit()
            toolbar.title = title
        } else {
            pageContent = supportFragmentManager.getFragment(savedInstanceState, KEY_FRAGMENT!!)
            title = savedInstanceState.getString(KEY_TITLE)
            supportFragmentManager.beginTransaction().replace(R.id.fragment_container, pageContent!!).commit()
            toolbar.title = title
        }
    }

    override fun onSaveInstanceState(outState: Bundle) {
        outState.putString(KEY_TITLE, title)
        supportFragmentManager.putFragment(outState, KEY_FRAGMENT!!, pageContent!!)
        super.onSaveInstanceState(outState)
    }

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(R.menu.main_menu, menu)
        return super.onCreateOptionsMenu(menu)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.action_change_settings) {
            val mIntent = Intent(Settings.ACTION_LOCALE_SETTINGS)
            startActivity(mIntent)
        }
        return super.onOptionsItemSelected(item)
    }

    companion object {
        fun getLaunchIntent(from: Context) = Intent(from, MainActivity::class.java).apply {
            addFlags(Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_CLEAR_TASK)
        }
    }

}